tarorpc/taro.proto

```diff
@@ -592,8 +598,39 @@ message ProofFile {
     string genesis_point = 2;
 }
 
-message ProofVerifyResponse {
+message DecodedProof {
+    // The index of the decoded proof.
+    uint64 proof_index = 1;
+
+    // The total number of proofs contained in the raw proof.
+    uint64 number_of_proofs = 2;
+
+    Asset asset = 3;
+
+    bytes tx_merkle_proof = 4;
+    bytes inclusion_proof = 5;
+    repeated bytes exclusion_proofs = 6;
+}
+
+message VerifyProofResponse {
     bool valid = 1;
+    DecodedProof decoded_proof = 2;
+}
+
+message DecodeProofRequest {
+    // The raw proof in bytes to decode, which may contain multiple proofs.
+    bytes raw_proof = 1;
+
+    oneof filter {
```

Comment:
we can avoid this complex field if we use the default value in our favour. Given that the default value is `0` while most of the time the endpoint will be used to decode the latest proof (which matches what the api should do even if I use it without reading the docs) what do you think about having the field `proofAtDepth` instead of `index`?

`marshalProofFile` needs to then transform this to an index with `index := proofFile.NumProofs() - 1 - proofAtDepth`, if that value is negeative it should fail. If I want to genesis proof and there are 5 proofs I need to used `proofAtDepth=4`

By default `proofAtDepth` will be 0, which means decodes the latest proof. It's a similar concept of what happens with git commits or block confirmations.

Reply:
I like that, makes the API more self-explanatory . Great idea!